packages/js/src/feeds/feeds.ts

```diff
@@ -1,33 +1,187 @@
-import type { PaginatedResponse } from '../types';
-import { BaseModule } from '../utils/base-module';
+import { BaseModule } from '../base-module';
 import { Notification } from './notification';
-
-interface FetchFeedOptions {
-  page?: number;
-  feedIdentifier?: string | string[];
-  seen?: boolean;
-  read?: boolean;
-  limit?: number;
-  payload?: Record<string, unknown>;
-}
+import { PaginatedResponse, NotificationStatus, TODO } from '../types';
+import type {
+  FetchFeedArgs,
+  FetchCountArgs,
+  MarkNotificationAsArgs,
+  MarkAllNotificationsAsArgs,
+  RemoveNotificationArgs,
+  RemoveAllNotificationsArgs,
+  MarkNotificationActionAsArgs,
+  MarkNotificationsAsArgs,
+  RemoveNotificationsArgs,
+  MarkNotificationAsByIdArgs,
+  MarkNotificationAsByInstanceArgs,
+  MarkNotificationsAsByIdsArgs,
+  MarkNotificationsAsByInstancesArgs,
+  RemoveNotificationByIdArgs,
+  RemoveNotificationByInstanceArgs,
+  RemoveNotificationsByIdsArgs,
+  RemoveNotificationsByInstancesArgs,
+  MarkNotificationActionAsByIdArgs,
+  MarkNotificationActionAsByInstanceArgs,
+} from './types';
+import { READ_OR_UNREAD, SEEN_OR_UNSEEN } from '../utils/notification-utils';
+import { markActionAs, markNotificationAs, markNotificationsAs, remove, removeNotifications } from './helpers';
 
 export class Feeds extends BaseModule {
-  async fetch({ page = 1, ...restOptions }: FetchFeedOptions): Promise<PaginatedResponse<Notification>> {
+  async fetch({ page = 0, status, ...restOptions }: FetchFeedArgs = {}): Promise<PaginatedResponse<Notification>> {
     return this.callWithSession(async () => {
+      const args = { page, status, ...restOptions };
       try {
-        this._emitter.emit('feeds.fetch.pending');
+        this._emitter.emit('feeds.fetch.pending', { args });
 
-        const response = await this._apiService.getNotificationsList(page, restOptions);
+        const response = await this._apiService.getNotificationsList(page, {
+          ...restOptions,
+          ...(status && SEEN_OR_UNSEEN.includes(status) && { seen: status === NotificationStatus.SEEN }),
```

Comment:
This conversion of status between the SDK and the apiService seems unnecessary. Is it mandatory to keep the apiService backward compatible?

Reply:
Unfortunately, it is a public package :( but I think that we would need to deprecate it and suggest using the new SDK instead